lib/go-tc/servers.go

```diff
@@ -1101,6 +1105,46 @@ func (s *ServerV40) ToServerV2FromV4() (ServerNullableV2, error) {
 	return legacyServer, nil
 }
 
+// ServerUpdateStatusV4 is the type of each entry in the `response` property of
+// the response from Traffic Ops to GET requests made to its
+// /servers/{{host name}}/update_status in the latest minor API
+// v4.0 endpoint.
+type ServerUpdateStatusV4 ServerUpdateStatusV40
+
+// ServerUpdateStatusV40 is the type of each entry in the `response` property of
+// the response from Traffic Ops to GET requests made to its
+// /servers/{{host name}}/update_status in API v4.0 endpoint.
+type ServerUpdateStatusV40 struct {
+	HostName             string     `json:"host_name"`
+	UpdatePending        bool       `json:"upd_pending"`
+	RevalPending         bool       `json:"reval_pending"`
+	UseRevalPending      bool       `json:"use_reval_pending"`
+	HostId               int        `json:"host_id"`
+	Status               string     `json:"status"`
+	ParentPending        bool       `json:"parent_pending"`
+	ParentRevalPending   bool       `json:"parent_reval_pending"`
+	ConfigUpdateTime     *time.Time `json:"config_update_time"`
+	ConfigApplyTime      *time.Time `json:"config_apply_time"`
+	RevalidateUpdateTime *time.Time `json:"revalidate_update_time"`
+	RevalidateApplyTime  *time.Time `json:"revalidate_apply_time"`
```

Comment:
Ok, small bit of history. I had originally designed this to have a separate database table for these timestamp values. The thought was that not all servers are cache servers, and thus not all servers needed these values. This initially led me to make these nullable.

The other dilemma I have here, and I think we will touch on this later in the review as well, is that the `ServerV40` struct is used for both sending and receiving in the CRUD calls, as well as elsewhere I presume. If I make these required fields, then those sending POSTs requests would have to include values for these fields.

If memory servers, the current logic requires sending `upd_pending` boolean, thus when creating a server you would have to know that a config update must be applied (or not) at creation of the server. To keep with this logic, I left these nullable allowing the client to send what they _know_ they would need, instead of being forced to potentially set values for timestamp fields not quite understanding the ramifications. Granted a user could still run into this issue if they chose to set these fields, but currently it's not required.